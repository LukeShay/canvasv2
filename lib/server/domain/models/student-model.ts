/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable global-require */
import { Model, RelationMappings, JSONSchema } from 'objection';
import Constants from '../constants';

export class StudentModel extends Model {
  static tableName = Constants.STUDENTS_TABLE;

  static idColumn = 'id';

  static get jsonSchema(): JSONSchema {
    return {
      properties: {
        classId: { type: 'string' },
        studentId: { type: ['string'] },
      },
      required: ['studentId', 'classId'],
      type: 'object',
    };
  }

  static get relationMappings(): RelationMappings {
    // ts-expect-error
    const { UserModel } = require('./user-model');
    // ts-expect-error
    const { ClassModel } = require('./class-model');

    return {
      class: {
        join: {
          from: `${StudentModel.tableName}.classId`,
          to: `${ClassModel.tableName}.${ClassModel.idColumn}`,
        },
        modelClass: ClassModel,
        relation: Model.BelongsToOneRelation,
      },
      student: {
        join: {
          from: `${StudentModel.tableName}.studentId`,
          to: `${UserModel.tableName}.${UserModel.idColumn}`,
        },
        modelClass: UserModel,
        relation: Model.BelongsToOneRelation,
      },
    };
  }
}
