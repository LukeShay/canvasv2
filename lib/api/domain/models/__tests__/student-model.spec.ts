import { StudentModel } from '../student-model';

describe('StudentModel', () => {
  it("should have idColum 'id'", () => {
    expect(StudentModel.idColumn).toEqual('id');
  });

  it('should should have jsonSchema that matches snapshot', () => {
    expect(StudentModel.jsonSchema).toMatchSnapshot();
  });

  it('should should have relationMappings that matches snapshot', () => {
    expect(StudentModel.relationMappings).toMatchSnapshot();
  });
});
