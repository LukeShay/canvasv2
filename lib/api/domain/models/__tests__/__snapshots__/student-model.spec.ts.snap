// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentModel should should have jsonSchema that matches snapshot 1`] = `
Object {
  "properties": Object {
    "classId": Object {
      "type": "string",
    },
    "studentId": Object {
      "type": Array [
        "string",
      ],
    },
  },
  "required": Array [
    "studentId",
    "classId",
  ],
  "type": "object",
}
`;

exports[`StudentModel should should have relationMappings that matches snapshot 1`] = `
Object {
  "class": Object {
    "join": Object {
      "from": "Students.classId",
      "to": "Classes.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
  "student": Object {
    "join": Object {
      "from": "Students.studentId",
      "to": "Users.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
}
`;
